src/calendar/index.tsx

```diff
@@ -70,28 +127,21 @@ class Calendar extends Component {
     /** Allow rendering of a totally custom header */
     customHeader: PropTypes.any
   };
-
   static defaultProps = {
     enableSwipeMonths: false
   };
 
-  constructor(props) {
-    super(props);
-
-    this.style = styleConstructor(props.theme);
-
-    this.state = {
-      currentMonth: props.current ? parseDate(props.current) : XDate()
-    };
-
-    // this.shouldComponentUpdate = shouldComponentUpdate;
-  }
+  state = {
+    currentMonth: this.props.current ? parseDate(this.props.current) : new XDate()
+  };
+  style = styleConstructor(this.props.theme);
+  header: RefObject<CalendarHeader> = React.createRef();
 
-  addMonth = count => {
-    this.updateMonth(this.state.currentMonth.clone().addMonths(count, true));
+  addMonth = (count: number) => {
+    this.updateMonth(this.state.currentMonth.clone().addMonths(count, true), false);
   };
 
-  updateMonth = (day, doNotTriggerListeners) => {
+  updateMonth = (day: any, doNotTriggerListeners: boolean) => {
```

Comment:
You can set a default value here so you won't need to pass it on there places 
```suggestion
  updateMonth = (day: any, doNotTriggerListeners?: boolean = false) => {
```

Reply:
Done